hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -143,8 +144,16 @@ private void tryRemoveKey(Path path) {
 
     @Override
     public boolean delete(Path path, boolean recursive) throws IOException {
-        if (recursive) {
-            throw new UnsupportedOperationException("EncryptedFileSystem does not support recursive deletes");
+        if (recursive && isDirectory(path)) {
+            FileStatus[] children = listStatus(path);
+
+            for (FileStatus child : children) {
+                boolean deleteSuccessful = delete(child.getPath(), true);
+
+                if (!deleteSuccessful) {
+                    return false;
```

Comment:
The definition of `delete` in the [Hadoop documentation](http://hadoop.apache.org/docs/r2.7.5/api/org/apache/hadoop/fs/FilterFileSystem.html#delete-org.apache.hadoop.fs.Path-boolean-) do not specify whether we should stop immediately here, or continue with the rest of the delete. If we're modeling this after `rm -rf` then I'd expect it to continue deleting what it can though, right? Not sure if there's a reference impl elsewhere we can look at.

Reply:
Easy to do either, but yeah, I'm unsure what behavior is preferable. RawLocalFileSystem continues and eventually returns false if anything failed. FTPFileSystem continues and ignores any intermediate issues (maybe a bug). DistributedFileSystem is hard to follow, but seems to atomically destroy everything, making this is a non-issue. S3FileSystem matches this behavior, returning early if anything fails.

Reply:
I'd really prefer to avoid supporting recursive deletes within EncryptedFileSystem until we can do so efficiently. I'd propose that you do:
```
RemoteIterator<LocatedFileStatus> files = listFiles(path, true);
while (files.hasNext()) {
    delete(files.next().getPath(), false);
}
```
as a workaround for the time being.